job-runner/chain/chain.go

```diff
@@ -15,19 +15,29 @@ import (
 
 // chain represents a job chain and some meta information about it.
 type chain struct {
```

Comment:
I think it'd help to make this `Chain` so we can see all it's methods from a high level (via godoc).